src/components/types.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable import/prefer-default-export */
+import { arrayOf, string, bool, shape } from "prop-types";
+
+const LinkType = {
+  text: string,
+  href: string,
+  icon: shape({
+    type: string,
+    src: string
+  })
+};
+
+const IconType = {
+  type: string,
+  src: string
+};
+
+const AttractionType = {
+  id: string,
+  name: string,
+  seoUrl: string,
+  url: string,
+  classification: shape({
+    segment: shape({
+      name: string,
+      __typename: string
+    }),
+    __typename: string
+  }),
+  __typename: string
+};
+
+const AttractionsType = arrayOf(shape(AttractionType));
+
+const DateFormattedType = {
+  dateTitle: string,
+  dateSubTitle: string,
+  venueDateTime: string,
+  venueDateTimeFormatted: string,
+  __typename: string
+};
+
+const DatesType = {
+  spanMultipleDays: bool,
+  start: shape({
+    dateTBA: bool,
+    timeTBA: bool,
+    localDate: string,
+    dateTime: string,
+    noSpecificTime: bool,
+    __typename: string
+  }),
+  status: shape({
+    code: string,
+    __typename: string
+  }),
+  __typename: string
+};
+
+const ProductType = {
+  id: string,
+  name: string,
+  type: string,
+  url: string,
+  __typename: string
+};
+
+const ProductsType = arrayOf(shape(ProductType));
+
+const VenueType = {
+  id: string,
+  name: string,
+  seoLink: string,
+  postalCode: string,
+  address: shape({
+    line1: string,
+    __typename: string
+  }),
+  state: shape({
+    stateCode: string,
+    name: string,
+    __typename: string
+  }),
+  country: shape({
+    countryCode: string,
+    name: string,
+    __typename: string
+  }),
+  __typename: string
+};
+
+const EventType = {
```

Comment:
I feel like a lot of these shapes are too heavily premised on the data structures that our team uses. We should abstract away the business logic of transforming these objects away from these functional components. That will make these components more reusable across teams.

Reply:
+1